lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -1,15 +1,255 @@
+    {
+        get { return false; }
+    }
+
+    public override Size MaxPhysicalWindowSize
+    {
+        get { return new Size(165, 66); }
+    }
+
+    public override Size MaxWindowSize
+    {
+        get { return new Size(165, 66); }
+    }
+
+    public override void FlushInputBuffer()
+    {
+      throw new NotImplementedException();
+    }
+
+    public override BufferCell[,] GetBufferContents(Rectangle rectangle)
+    {
+      throw new NotImplementedException();
+    }
+
+    public override KeyInfo ReadKey(ReadKeyOptions options)
+    {
+      throw new NotImplementedException();
+    }
+
+    public override void ScrollBufferContents(Rectangle source, Coordinates destination, Rectangle clip, BufferCell fill)
+    {
+      throw new NotImplementedException();
+    }
+
+    public override void SetBufferContents(Rectangle rectangle, BufferCell fill)
+    {
+      throw new NotImplementedException();
+    }
+
+    public override void SetBufferContents(Coordinates origin, BufferCell[,] contents)
+    {
+      throw new NotImplementedException();
+    }
+  }
+
+  public class PowerShellHostUserInterface : PSHostUserInterface
+  {
+    private PowerShellRawUserInterface _rawui;
+    private StringBuilder _sb;
+
+    public PowerShellHostUserInterface()
+    {
+      _sb = new StringBuilder();
+    }
+
+    public override PSHostRawUserInterface RawUI
+    {
+      get
+      {
+        if ( _rawui == null){
+          _rawui = new PowerShellRawUserInterface();
+        }
+        return _rawui;
+      }
+    }
+
+    public override void Write(ConsoleColor foregroundColor, ConsoleColor backgroundColor, string value)
+    {
+      _sb.Append(value);
+    }
+
+    public override void Write(string value)
+    {
+      _sb.Append(value);
+    }
+
+    public override void WriteDebugLine(string message)
+    {
+      _sb.AppendLine("DEBUG: " + message);
+    }
+
+    public override void WriteErrorLine(string value)
+    {
+      _sb.AppendLine(value);
+    }
+
+    public override void WriteLine(string value)
+    {
+      _sb.AppendLine(value);
+    }
+
+    public override void WriteVerboseLine(string message)
+    {
+      _sb.AppendLine("VERBOSE: " + message);
+    }
+
+    public override void WriteWarningLine(string message)
+    {
+      _sb.AppendLine("WARNING: " + message);
+    }
+
+    private bool hasLoggedStartProgress = false;
+    private bool hasLoggedFinalProgress = false;
+    public override void WriteProgress(long sourceId, ProgressRecord record)
+    {
+      if(record.PercentComplete == -1) return;
+      if(hasLoggedFinalProgress) return;
+      if(!hasLoggedStartProgress)
+      {
+        hasLoggedStartProgress = true;
+        _sb.AppendLine(record.Activity.ToString());
+      }
+
+      // http://stackoverflow.com/a/888569/18475
+      var msg = string.Format("\rProgress: {0}% - {1}", record.PercentComplete.ToString(), record.StatusDescription.PadRight(50, ' ') );
+      _sb.Append(msg);
```

Comment:
What does this actually look like in a log?  Seems like it could generate a ton of output when progress logs in `1%` increments (i.e. that would generate 100 lines of mostly useless text).  Do we care?